lib/parser/livesplit_core_parser.rb

```diff
@@ -26,24 +26,37 @@ def self.parse(run, fast: false)
     end
 
     run = parse_result.unwrap
+    run_metadata = run.metadata
 
     run_object = {
       program: program,
       game: run.game_name,
+      game_icon: run.game_icon.presence,
       category: run.category_name,
-      srdc_id: run.metadata.run_id,
+      name: run.extended_name(false),
+      metadata: {
+        srdc_id: run_metadata.run_id.presence,
```

Comment:
What does .presence do?

Reply:
If the object is `present?`, then it returns the object otherwise it returns `nil`.  I put it there to change the empty strings to `nil`s so we could just do `.nil?` checks to try and standardize what checks we perform.

Here is the api dock link for the method: https://apidock.com/rails/Object/presence

Reply:
Oh, that's kind of what I figured, but the documentation (which I found too) doesn't mention the empty string to nil conversion, so I wasn't sure.

Reply:
Ah gotcha. I used [this](https://stackoverflow.com/questions/885414/a-concise-explanation-of-nil-v-empty-v-blank-in-ruby-on-rails/20663389#20663389) table that glacials gave me forever ago so I know what values are blank and present and stuff.